protocols/kad/src/behaviour/test.rs

```diff
@@ -1287,3 +1287,87 @@ fn network_behaviour_inject_address_change() {
         kademlia.addresses_of_peer(&remote_peer_id),
     );
 }
+
+#[test]
+fn client_mode() {
+    let mut cfg = KademliaConfig::default();
+    cfg.set_mode(Mode::Client);
+
+    // Create a server and client peer.
+    let client_swarm = build_node_with_config(cfg);
+    let server_swarm = build_node();
+
+    let mut swarms = Vec::new();
+    swarms.push(client_swarm.1);
+    swarms.push(server_swarm.1);
+
+    let mut swarm_addrs = Vec::new();
+    swarm_addrs.push(client_swarm.0);
+    swarm_addrs.push(server_swarm.0);
+
+    let swarm_ids: Vec<_> = swarms
+        .iter()
+        .map(|swarm| swarm.local_peer_id())
+        .cloned()
+        .collect();
+
+    block_on(poll_fn(move |ctx| {
+        // flags to make sure both peers are listening
+        let mut client_listening = false;
+        let mut server_listening = false;
+
+        for swarm in swarms.iter_mut() {
+            loop {
+                match swarm.poll_next_unpin(ctx) {
+                    Poll::Ready(Some(SwarmEvent::NewListenAddr {
+                        listener_id: _,
+                        address,
+                    })) => {
+                        if address == swarm_addrs[0] {
+                            client_listening = true;
+                            return Poll::Ready(());
+                        }
+                        if address == swarm_addrs[1] {
+                            server_listening = true;
+                            return Poll::Ready(());
+                        }
+                    }
+                    Poll::Ready(_) => {
```

Comment:
How about waiting for a `KademliaEvent::RoutingUpdated` from the client swarm with the server peer id?